x-pack/plugins/apm/public/components/fleet_integration/apm_agents/agent_instructions_mappings.ts

```diff
@@ -35,12 +48,14 @@ export const ApmAgentInstructionsMappings: Array<{
   title: string;
   variantId: string;
   createAgentInstructions: CreateAgentInstructions;
+  AgentRuntimeAttachment?: (props: AgentRuntimeAttachmentProps) => ReactElement;
```

Comment:
Probably should be `ComponentType` since `ReactElement` is restrictive (doesn't allow `null`) and `ComponentType` can be a class or function component.

```suggestion
  AgentRuntimeAttachment?: ComponentType<AgentRuntimeAttachmentProps>
```
```suggestion
  AgentRuntimeAttachment?: (props: AgentRuntimeAttachmentProps) => ReactElement;
```